### DISCUSSION TITLE
ModMenu reverse bridge? (or ffapi port)

### DISCUSSION THREAD

**Discussion by sleepingdragoninn** [Sinytra/ConnectorExtras#50] [Ideas]
So ModMenu works quite well out of the box with connector - mods are shown, square icons all work, and metadata is all nicely intact.

![image](https://github.com/user-attachments/assets/b6d51b41-d63d-429b-aff9-ba860cc22a9c)

![image](https://github.com/user-attachments/assets/c6c2017d-de64-49c3-b416-24345bc041c9)

The only downsides with using it i see are:
- Non-square icons aren't cropped to be square 
- The neo mod menu icon appears below it
- neo config screens aren't added to mods
- issue tracker and display URLs aren't passed through to fabric mod containers (sounds like an FFL bug)

Adding connextras hides the mod menu button and makes mod menu config screens work on the neo menu, so my question is - what would be the most effective way to have a _completely working_ version of modmenu on a neo 1.21 + connector environment?

I could certainly imagine a config switch in connextras that hides the forge mod menu button instead, and the same config screen bridging happening in reverse to make native neo configs appear on modmenu - maybe even a workaround for non-square icons. 

But is that within scope for connector extras? Or would it be better served by an explicit port of mod menu? And if there was an explicit port of mod menu, how could it be set up to ensure that fabric mod metadata isn't experiencing losses by going through the FFL translation layer (things like author and contributor arrays)?

**Comment by 1foxy2** [comment]
i am currently working on modmenu port for neoforge which is compatible with connector and fabric metadata https://github.com/1foxy2/neo_mod_menu, nearly finished just need to change the modid so it will be compatible with extras
![image](https://github.com/user-attachments/assets/f7af1c30-2267-4550-b3a8-3b8a4b388adf)
